src/session_p.h

```diff
@@ -72,13 +83,71 @@ struct nc_client_ssh_opts {
     char *username;
 };
 
+struct nc_certificate {
+    char *name;
+    char *cert_data;
+};
```

Comment:
For simple structures that are not used on several places I find it better to put the struct inline into the other struct.

Reply:
I will rearrange this structure later, when using certificates is supported.